test/test-url.cpp

```diff
@@ -247,6 +247,228 @@ TEST(ParseTest, TestPortNotNumber)
     ASSERT_THROW(Url::Url("http://www.python.org:80hello/"), Url::UrlParseException);
 }
 
+TEST(AssignTest, AssignsValue)
+{
+    Url::Url assignee("");
+    Url::Url parsed("http://user@example.com:8080/path;param?query#fragment");
+    assignee.assign(parsed);
+    EXPECT_EQ(assignee.scheme(), parsed.scheme());
+    EXPECT_EQ(assignee.userinfo(), parsed.userinfo());
+    EXPECT_EQ(assignee.host(), parsed.host());
+    EXPECT_EQ(assignee.port(), parsed.port());
+    EXPECT_EQ(assignee.path(), parsed.path());
+    EXPECT_EQ(assignee.params(), parsed.params());
+    EXPECT_EQ(assignee.query(), parsed.query());
+    EXPECT_EQ(assignee.fragment(), parsed.fragment());
+}
+
+TEST(EqualityTest, RequiresSchemeEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setScheme("https");
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresUserinfoEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setUserinfo("name");
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresHostEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setHost("new.host");
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresPortEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setPort(1234);
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresPathEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setPath("/new/path");
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresParamsEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setParams("new;params");
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresQueryEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setQuery("new=query");
+    EXPECT_NE(a, b);
+}
+
+TEST(EqualityTest, RequiresFragmentEquality)
+{
+    Url::Url a("http://user@example.com:8080/path;param?query#fragment");
+    Url::Url b(a);
+    EXPECT_EQ(a, b);
+    b.setFragment("new-fragment");
+    EXPECT_NE(a, b);
+}
+
+TEST(EquivTest, HttpPort)
+{
+    Url::Url a("http://foo.com:80");
+    Url::Url b("http://foo.com/");
+    EXPECT_EQ(true, a.equiv(b));
```

Comment:
Thanks -- I was too lazy to look up whether there was such an expect last night, but updated.